core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java

```diff
@@ -1130,18 +1148,28 @@ public long saveRun(Run run) throws IOException {
   @Override
   public void saveRuns(Collection<Run> runs) throws IOException {
     if (runStore != null) {
-      runStore.saveAll(runs);
+      List<Run> newRuns = new ArrayList<>();
+      List<Run> savedRuns = new ArrayList<>();
+      for (Run run : runs) {
+        if (run.getId() == AbstractRun.UNSAVED_ID) {
+          newRuns.add(run);
+        } else {
+          savedRuns.add(run);
+        }
+      }
+      runStore.saveAll(newRuns);
+      for (Run run : savedRuns) {
+        saveRun(run);
+      }
     } else {
       throw new IOException("No runStore available. Check that it has been declared in the Spring config.");
     }
   }
 
   @Override
-  public long saveRunQC(RunQC runQC) throws IOException {
+  public void saveRunQC(RunQC runQC) throws IOException {
```

Comment:
It's easily accessible, can be useful, and all of the other save methods return the inserted ID. I don't see a reason to remove it, and would prefer consistency